.eslintrc

```
@@ -5,25 +5,83 @@
 		"es6": true,
 		"jest": true
 	},
+	"parser": "@typescript-eslint/parser",
 	"parserOptions": {
 		"ecmaFeatures": {
-			"experimentalObjectRestSpread": true,
 			"jsx": true
 		}
 	},
 	"extends": "liferay",
-	"plugins": ["notice", "no-for-of-loops", "react"],
+	"plugins": [
+		"notice",
+		"no-for-of-loops",
+		"react",
+		"react-hooks",
+		"sort-destructure-keys",
+		"sort-imports-es6-autofix"
+	],
 	"rules": {
+		"comma-dangle": [2, "always"],
+		"default-case": 2,
+		"eol-last": [2, "never"],
+		"max-len": 0,
+		"no-console": 2,
 		"no-for-of-loops/no-for-of-loops": 2,
-		"no-unused-vars": ["error", {"ignoreRestSiblings": true}],
-		"react/jsx-uses-react": "error",
-		"react/jsx-uses-vars": "error",
 		"no-mixed-spaces-and-tabs": ["error", "smart-tabs"],
+		"no-return-assign": [2, "always"],
+		"no-undef": 2,
+		"no-underscore-dangle": [
+			2,
+			{
+				"allowAfterThis": true,
+				"enforceInMethodNames": true
+			}
+		],
+		"no-unused-vars": [
+			2,
+			{
+				"args": "after-used",
+				"ignoreRestSiblings": false,
+				"vars": "all"
+			}
+		],
 		"notice/notice": [
 			"error",
 			{
-				"templateFile": "copyright.js"
+				"template": "/**\n * © 2018 Liferay, Inc. <https://liferay.com>\n *\n * SPDX-License-Identifier: BSD-3-Clause\n */\n"
```

hey @bryceosterhaus, just remembering that the year should start in 2019. 🙂
```suggestion
				"template": "/**\n * © 2019 Liferay, Inc. <https://liferay.com>\n *\n * SPDX-License-Identifier: BSD-3-Clause\n */\n"
```

---

So I was reading through the original requirements for licensing in this comment, https://github.com/liferay/clay/pull/1451#issuecomment-456384580

This portion stuck out to me

> just change the year to whatever year that file was in (if that’s too much work, `2014` as the year of the first commit should do). And then do _not_ bother with updating the year in the future. When you create new files, just put the year when you created it – e.g. for new files started this year, that would be 2019.

And I just realized with using the eslint rule, it will update the year everywhere. It sounds like the requirements for licensing is to add the year the file is created, but then to not update the year in subsequent years.

I'll update everything to 2019 since its the first time we are adding this header, but I'm gonna look at the eslint rule and see if we can configure it to not update in subsequent years.

---

@matuzalemsteles just updated pr. The eslint plugin actually allows a `<%= YEAR %>` param in the template. This should solve the issue of new files and existing files with licenses. It only updates to the current year if its a new file.

---

Oh yeah, great @bryceosterhaus! Thanks